routes/users.js

```diff
@@ -1,33 +1,58 @@
 const express = require('express');
 const donationData = require('../data/donations');
+const userData = require('../data/users');
+
 const router = express.Router();
+const authMiddleWare = require('../middlewares/auth');
 
-router.get('/admin/:id', async (req, res) => {
-  let donations = await donationData.allDonations();
-  res.status(200).render('users/admin/dashboard', {
-    title: 'Admin',
-    showApproveReject: true,
-    donations,
-    message: req.flash(),
-  });
-});
+router.get('/:id/dashboard', authMiddleWare.loginRequired, async (req, res) => {
+  try {
+    let sessionUser = req.session.user;
+    // search for user from id given in params
+    let searchedUser = await userData.getUserById(req.params.id);
+    if (searchedUser.role_id != sessionUser.role_id) throw 'Invalid User';
```

Comment:
then find and check if the correct user is set in the session and role is also correct.